universe/assets_3d/api_assets_3d.go

```diff
@@ -0,0 +1,189 @@
+package assets_3d
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/momentum-xyz/ubercontroller/universe"
+	"github.com/momentum-xyz/ubercontroller/universe/api"
+	"github.com/momentum-xyz/ubercontroller/universe/api/dto"
+	"github.com/momentum-xyz/ubercontroller/utils"
+)
+
+func (a *Assets3d) apiGetAssets3d(c *gin.Context) {
+	queryParams := struct {
+		kind string `form:"kind" json:"kind"`
+	}{}
+
+	if err := c.ShouldBind(&queryParams); err != nil {
+		err = errors.WithMessage(err, "Assets3d: apiGetAssets3d: failed to bind query parameters")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_request_query", err, a.log)
+	}
+
+	a3dMap := make(map[uuid.UUID]universe.Asset3d)
+	predicateFn := func(asset3dID uuid.UUID, asset3d universe.Asset3d) bool {
+		entry := asset3d.GetEntry()
```

Comment:
if you need only meta, then it's better to get it instead of getting whole enty
keep in mind that when you are getting entry it's creating new object for you